src/seismometer/data/pandas_helpers.py

```diff
@@ -153,11 +175,132 @@ def infer_label(dataframe: pd.DataFrame, label_col: str, time_col: str) -> pd.Da
         dataframe[label_col] = dataframe[label_col].astype(float)
     except BaseException:  # Leave as nonnumeric
         pass
-    dataframe.loc[dataframe[label_col].isna(), label_col] = (
-        dataframe.loc[dataframe[label_col].isna(), time_col].notna().astype(int)
+
+    dataframe.loc[dataframe[label_col].isna(), label_col] = np.where(
+        dataframe.loc[dataframe[label_col].isna(), time_col].notna(), (impute_val or 1), 0
     )
+    dataframe[label_col].replace({"1": 1, "0": 0}, inplace=True)
+
     return dataframe
 
+def _merge_event_counts(
+    left: pd.DataFrame,
+    right: pd.DataFrame,
+    pks: list[str],
+    event_name: str,
+    event_label: str,
+    window_hrs: Optional[Number] = None,
+    min_offset: Number = 0,
+    l_ref: str = "Time",
+    r_ref: str = "Time",
+) -> pd.DataFrame:
+    """Creates a new column for each event in the right frame's event_label column, 
+    counting the number of times that event has occurred"""
+
+    if (N := right[event_label].nunique()) > MAXIMUM_NUM_COUNTS:
+        logger.warning(f"Maximum number of unique events to count is {MAXIMUM_NUM_COUNTS}, but {N} were found for {event_name}. Only the first {MAXIMUM_NUM_COUNTS} will be counted.")
+        #Filter right frame to the only contain the first MAXIMUM_NUM_COUNTS events
+        events_to_count = right[event_label].unique()[:MAXIMUM_NUM_COUNTS]
+        right = right[right[event_label].isin(events_to_count)]
+
+    event_name_map = {event: event_value_count(str(event)) for event in right[event_label].unique()} #Create dictionary to map column names with
+
+    if window_hrs is not None:
```

Comment:
You're right, I've moved the windowing so that it's the first thing to happen in _merge_event_counts.

https://github.com/epic-open-source/seismometer/blob/4909413838d4a97c0ff81e3de0c5776d1d833642/src/seismometer/data/pandas_helpers.py#L198-L202

[I also said this in my other reply](https://github.com/epic-open-source/seismometer/pull/76#discussion_r1735190073), but I've also gone ahead and changed the selection criteria to be based on number of appearances instead of order of order of appearance.